pkg/resource/deploy/step_generator_test.go

```diff
@@ -360,3 +360,105 @@ func TestEngineDiff(t *testing.T) {
 		})
 	}
 }
+
+func TestResourceInheritsOptionsFromParent(t *testing.T) {
+	t.Parallel()
+
+	t.Run("children missing DeletedWith should inherit DeletedWith", func(t *testing.T) {
```

Comment:
minor: Test name is part of the subtest name so we can eliminate some redundancy in the name.
Longer explanations can be inside comments if needed.

For these tests, maybe: DeletedWith/inherit, DeletedWith/override, DeletedWith/empty?
(The "/" will group them all under an implicit "DeletedWith" test group.)